torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -273,11 +276,12 @@ void PropagateShapeOnNode(Node * node, bool insert_expands) {
   // For expensive ops we can directly encode their shape propagation
   // here, otherwise we fallback to running a fake version of the op
   // to get a quick and dirty propagation.
-  if (insert_expands && (
-      node->matches("aten::add(Tensor self, Tensor other, *, Scalar alpha) -> Tensor") ||
+  if (node->matches("aten::add(Tensor self, Tensor other, *, Scalar alpha) -> Tensor") ||
       node->matches("aten::sub(Tensor self, Tensor other, *, Scalar alpha) -> Tensor") ||
-      node->matches("aten::mul(Tensor self, Tensor other) -> Tensor") ||
-      node->matches("aten::div(Tensor self, Tensor other) -> Tensor") ||
+      node->matches("aten::mul(Tensor self, Tensor other) -> Tensor")) {
+    PropagateShapeOnNodeByRunningIt(node);
+    return;
+  } else if (insert_expands && (
```

Comment:
Why is only such a small subset of nodes excluded from the `insert_expands` logic?

Reply:
Because these are the only nodes that handle expands internally